### PR TITLE
[WFLY-15696] Start Faces 4 integration

### PR DESCRIPTION
**Pull request by jasondlee**
https://issues.redhat.com/browse/WFLY-15696

Bump JSF artifact versions
Add parent pom for jsf transform
Fork classes that need manual import changes
Disable some tests in testsuite/integration/basic that are no longer for Faces 4
Extract FacesAnnotation enum
Override enum in source-transform
Delete removed annotation refs from enum in source-transform

### CODE REVIEW

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/unsupportedwar/web.xml [hunk 1]
```diff
@@ -15,16 +15,16 @@
     limitations under the License.
 -->
 <web-app version="4.0"
-    xmlns="http://xmlns.jcp.org/xml/ns/javaee"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd">
+         xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_4_0.xsd">
```
Comment (boris-unckel) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/unsupportedwar/web.xml:20]: This should be version 6 (line 17) and the jakarta namespace. Still a draft: https://jakarta.ee/xml/ns/jakartaee/#10

Comment (jasondlee) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/unsupportedwar/web.xml:20]: Noted, but I wouldn't spend too much on this just yet, as this is very much a work in progress. I just created the draft so I could easily see the changes in one place. :)

Comment (boris-unckel) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/unsupportedwar/web.xml:20]: Consider it NIT :-) And *I* learn from 99% of the rest ;-)

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/duplicateid/DuplicateIDIntegrationTestCase.java [hunk 2]
```diff
@@ -21,34 +32,31 @@
...
  */
 @RunWith(Arquillian.class)
 @RunAsClient
 public class DuplicateIDIntegrationTestCase {
 
+    // TODO: Disable this test for now. The inclusion of javax.* strings in static resources is troublesome, but we can
+    // fix that once main is switched to EE 10 and we no longer need to support
+    @BeforeClass
+    public static void beforeClass() {
+        assumeTrue(System.getProperty("ts.ee9") == null);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/duplicateid/DuplicateIDIntegrationTestCase.java:57]: Also ts.bootable.ee9

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/managedbean/annotation/AnnotatedManagedBeanTestCase.java [hunk 3]
```diff
@@ -37,6 +40,11 @@
 @RunWith(Arquillian.class)
 public class AnnotatedManagedBeanTestCase {
 
+    @BeforeClass
+    public static void beforeClass() {
+        assumeTrue(System.getProperty("ts.ee9") == null);
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/managedbean/annotation/AnnotatedManagedBeanTestCase.java:45]: Also ts.bootable.ee9

Comment (jasondlee) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/managedbean/annotation/AnnotatedManagedBeanTestCase.java:45]: Noted. Thanks. FWIW, I stole that method from existing classes. This kind of thing belongs in a test utility somewhere, but my quick scan didn't find one.

Comment (jasondlee) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/managedbean/annotation/AnnotatedManagedBeanTestCase.java:45]: Found in testsuite/shared. Extracted updated. Changes inbound.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/duplicateid/DuplicateIDIntegrationTestCase.java (hunk 2 above)
Comment (jasondlee) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jsf/duplicateid/DuplicateIDIntegrationTestCase.java]: Fixed

**Review by bstansberry — APPROVED** [review]

**Pull request merged**